common/stacks/src/main/scala/weco/api/stacks/http/SierraSource.scala

```diff
@@ -31,33 +39,82 @@ class SierraSource(client: HttpClient with HttpGet with HttpPost)(
   private implicit val umItemStub: Unmarshaller[HttpEntity, SierraItemData] =
     CirceMarshalling.fromDecoder[SierraItemData]
 
+  private implicit val umItemEntriesStub
+    : Unmarshaller[HttpEntity, SierraItemDataEntries] =
+    CirceMarshalling.fromDecoder[SierraItemDataEntries]
+
   private implicit val umErrorCode: Unmarshaller[HttpEntity, SierraErrorCode] =
     CirceMarshalling.fromDecoder[SierraErrorCode]
 
+  type Unmarshalled[T] =
+    Future[Either[SierraItemLookupError.ItemHasNoStatus, T]]
+
+  def unmarshalOKResponse[T](
+    response: model.HttpResponse
+  )(implicit um: Unmarshaller[model.HttpResponse, T]): Unmarshalled[T] =
+    Unmarshal(response)
+      .to[T]
+      .map(Right(_))
+      .recover {
+        case t: Throwable =>
+          Left(SierraItemLookupError.ItemHasNoStatus(t))
+      }
```

Comment:
I wonder if a pattern match would be neater than a series of `map … recover` here?